SimG4Core/CustomPhysics/plugins/RHStopTracer.cc

```diff
@@ -55,7 +55,10 @@ void RHStopTracer::update (const BeginOfEvent * fEvent) {
 void RHStopTracer::update (const BeginOfTrack * fTrack) {
   const G4Track* track = (*fTrack)();
   const G4ParticleDefinition* part = track->GetDefinition();
-  if(part && part == mParticle &&  track->GetKineticEnergy() > mTraceEnergy) {
+  std::string stringPartName = part->GetParticleName();
```

Comment:
hi @jalimena @civanch - is there not a const & interface for GetParticleName?  (and the rePartName should become a member data variable..

but more generically - this really should be something other than strings and regexs for each track..  There is not a "number" you can use instead?

Reply:
@jalimena , @davidlange6 , yes, comparison of string in run time is not optimal and not needed. It is why in the previous version of the code "mParticle" was added. If we try to improve the code, in the class constructor this mParticle may be set to nullptr, in the update(const BeginOfEvent * fEvent) it is possible to add
if(!mParticle) mParticle = G4ParticleTable::GetParticleTable()->FindParticle(mTraceParticleName); 

This should not be done earlier (for example, in class constructor) because particle table is not filled.
After that in 

void RHStopTracer::update (const BeginOfTrack * fTrack)

instead of current lines 58-61 it would be enough to have 
if((part == mParticle) ||  track->GetKineticEnergy() > mTraceEnergy) {

Reply:
@jalimena , also why this condition:  track->GetKineticEnergy() > mTraceEnergy is needed? What is the use case?